src/main/java/com/elytradev/concrete/resgen/ConcreteResourcePack.java

```diff
@@ -136,23 +136,21 @@ public static ResourceLocation nameToLocation(String name) {
 
 	@Override
 	protected InputStream getInputStreamByName(String name) throws IOException {
-		// Default to fallback if possible.
-		if (!((boolean) hasResourceName.invoke(realResourcePack, name))) {
-			if (cache.containsKey(name)) {
-				ConcreteLog.debug("ConcreteResourcePack was asked to obtain: {} using cache.", name);
-				return IOUtils.toInputStream(cache.get(name));
-			}
+		if (cache.containsKey(name)) {
+			ConcreteLog.debug("ConcreteResourcePack was asked to obtain: {} using cache.", name);
+			return IOUtils.toInputStream(cache.get(name));
+		}
 
-			ConcreteLog.debug("ConcreteResourcePack was asked to obtain: {}", name);
-			if (isLocation(name, "/blockstates/")) {
-				return IOUtils.toInputStream(generateBlockState(name));
-			} else if (isLocation(name, "/models/block/")) {
-				return IOUtils.toInputStream(generateBlockModel(name));
-			} else if (isLocation(name, "/models/item/")) {
-				return IOUtils.toInputStream(generateItemModel(name));
-			}
+		ConcreteLog.debug("ConcreteResourcePack was asked to obtain: {}", name);
+		if (isLocation(name, "/blockstates/")) {
+			return IOUtils.toInputStream(generateBlockState(name));
+		} else if (isLocation(name, "/models/block/")) {
+			return IOUtils.toInputStream(generateBlockModel(name));
+		} else if (isLocation(name, "/models/item/")) {
+			return IOUtils.toInputStream(generateItemModel(name));
 		}
 
+
 		// Use the real pack in the event that we're asked for a resource we don't have.
 		return (InputStream) getInputStreamByName.invoke(realResourcePack, name);
```

Comment:
Since this isn't a wrapper, `null` should be returned here so that the real resource pack is consulted directly.